### PR TITLE
chore: create safe redux selector hooks

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/25301

Establishes a pattern for selecting data from the redux store as safely as possible.

Introduces new hooks
`useUnsafeWalletSelector`
`useUnsafePanelSelector`
`useUnsafePageSelector`
`useSafeWalletSelector`
`useSafePanelSelector`
`useSafePageSelector`

The "safe" hooks will be used to select primitive values only `string | number | boolean | null | undefined` 
The "unsafe" hooks will be used to select objects and arrays from redux.
This "unsafe" label  will bring attention to areas where we are storing data that may fail redux's default comparison of a strict `===` reference comparison
## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
N/A, code cleanup

### CODE REVIEW

File: components/brave_wallet_ui/panel/container.tsx [hunk 1]
```diff
@@ -79,37 +80,33 @@ import { TransactionStatus } from '../components/extension/post-confirmation'
...
+  const selectedPanel = useSafePanelSelector(PanelSelectors.selectedPanel)
+  const addChainRequest = useUnsafePanelSelector(PanelSelectors.addChainRequest)
+  const signMessageData = useUnsafePanelSelector(PanelSelectors.signMessageData)
+  const switchChainRequest = useUnsafePanelSelector(PanelSelectors.switchChainRequest)
+  const suggestedTokenRequest = useUnsafePanelSelector(PanelSelectors.suggestedTokenRequest)
+  const getEncryptionPublicKeyRequest = useUnsafePanelSelector(PanelSelectors.getEncryptionPublicKeyRequest)
+  const decryptRequest = useUnsafePanelSelector(PanelSelectors.decryptRequest)
+  const connectingAccounts = useUnsafePanelSelector(PanelSelectors.connectingAccounts)
+  const selectedTransaction = useUnsafePanelSelector(PanelSelectors.selectedTransaction)
+  const hardwareWalletCode = useSafePanelSelector(PanelSelectors.hardwareWalletCode)
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/panel/container.tsx:109]: Might be nice to have these grouped by `safe` and `unsafe` as well as `wallet` and `panel`! :)

**Review by Douglashdaniel — COMMENTED** [review]
I like this flow!

File: components/brave_wallet_ui/panel/container.tsx (hunk 1 above)
Comment (josheleonard) [reply components/brave_wallet_ui/panel/container.tsx]: Updated. Grouped the selectors & added group comments

**Review by Douglashdaniel — APPROVED** [review]
LGTM!

**Pull request merged**